op-e2e/system/bridge/deposit_test.go

```diff
@@ -127,6 +116,178 @@ func TestDepositTxCreateContract(t *testing.T) {
 		byte(vm.PUSH1), 0x00,
 		byte(vm.SSTORE),
 	}
+)
+
+func TestMintCallToDelegatedAccount(t *testing.T) {
+	// This test:
+	// 1. Deploys a contract on L2 that stores the first word of the call data to storage
+	// 2. Sets the delegation designation for an account on L2 to the new contract
+	// 3. Sends a deposit to the account with the delegation code
+	// 4. Ensures the deposit properly calls the contract and stores the first word of the call data to storage
+	// 5. Ensures the deposit sender's nonce is incremented on L2
+	// 6. Ensures the deposit sender's balance is incremented on L2
+
+	op_e2e.InitParallel(t)
+	cfg := e2esys.DefaultSystemConfig(t)
+	cfg.DeployConfig = cfg.DeployConfig.Copy()
+	cfg.DeployConfig.ActivateForkAtGenesis(rollup.Isthmus)
+	delete(cfg.Nodes, "verifier")
+	sys, err := cfg.Start(t)
+	require.NoError(t, err, "Error starting up system")
+
+	l1Client := sys.NodeClient("l1")
+	l2Seq := sys.NodeClient("sequencer")
+
+	// Simple constructor that is prefixed to the actual contract code
+	// Results in the contract code being returned as the code for the new contract
+	deployData := append(deployPrefix, sstoreContract...)
+	signer := types.NewIsthmusSigner(cfg.L2ChainIDBig())
+
+	tx := types.MustSignNewTx(cfg.Secrets.Alice, signer, &types.DynamicFeeTx{
+		ChainID:   cfg.L2ChainIDBig(),
+		Nonce:     0,
+		To:        nil,
+		Value:     big.NewInt(0),
+		Gas:       500000,
+		GasFeeCap: big.NewInt(5000000000),
+		GasTipCap: big.NewInt(2),
+		Data:      deployData,
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	err = l2Seq.SendTransaction(ctx, tx)
+	cancel()
+	require.NoError(t, err, "Failed to send set code transaction")
+
+	ctx, cancel = context.WithTimeout(context.Background(), 15*time.Second)
+	r, err := wait.ForReceipt(ctx, l2Seq, tx.Hash(), 1)
+	cancel()
+	require.NoError(t, err, "Failed to get receipt for set code transaction")
+
+	contractAddr := crypto.CreateAddress(cfg.Secrets.Addresses().Alice, 0)
+	fmt.Println(r.Status, r.ContractAddress, contractAddr)
```

Comment:
Do you want to leave this in? Would `t.Log` be better?

Reply:
removed this! Thanks for catching!